nixos/modules/services/networking/zeronsd.nix

```diff
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.zeronsd;
+in {
+  options.services.zeronsd.enable = mkEnableOption (lib.mdDoc "ZeroNSD");
+
+  options.services.zeronsd.servedNetworks = mkOption {
+    default = [];
+    example = [ "a8a2c3c10c1a68de" ];
+    type = types.listOf types.str;
+    description = lib.mdDoc "List of ZeroTier Network IDs to start zeronsd instance for.";
+  };
+
+  options.services.zeronsd.domain = mkOption {
+    default = "home.arpa";
+    type = types.str;
+    description = lib.mdDoc ''
+      Domain under which zerotier records will be available
+    '';
+  };
+
+  options.services.zeronsd.apiTokenFile = mkOption {
+    default = null;
+    type = types.path;
+    description = lib.mdDoc ''
+      Path to a file containing the API Token for ZeroTier Central
+    '';
+  };
+
+  options.services.zeronsd.logLevel = mkOption {
+    default = "info";
+    type = types.str;
+    description = lib.mdDoc ''
+      Log Level (one of off, error, warn, info, debug, trace)
+    '';
+  };
+
+  options.services.zeronsd.wildcardMode = mkOption {
```

Comment:
why are these individual options and not part of a rfc42 style settings option/submodule?

Reply:
Refactored to use settings now ;)